src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1556,14 +1560,44 @@ public void EnsureLastEtagIsPersisted(DocumentsOperationContext context, long do
                     var tombstoneCollectionName = ExtractCollectionName(context, tombstoneCollection);
 
                     if (tombstoneCollectionName != collectionName)
+                    {
+                        Debug.Assert(false, "Should never happened after RavenDB-14325");
                         ThrowNotSupportedExceptionForCreatingTombstoneWhenItExistsForDifferentCollection(lowerId, collectionName, tombstoneCollectionName, e);
+                    }
                 }
 
                 throw;
             }
 
             return (newEtag, changeVector);
         }
+        private Slice ModifyLowerIdIfNeeded(DocumentsOperationContext context, Table table, Slice lowerId)
+        {
+            if (table.ReadByKey(lowerId, out _) == false)
+                return lowerId;
+
+            var length = lowerId.Content.Length;
+            Slice.From(context.Allocator, lowerId.Content.Ptr, length + sizeof(long) * 2, out var newLowerId); // TODO: dispose?
+
+            *(long*)(newLowerId.Content.Ptr + length) = ConflictedTombstoneIdMarkerLong;
+            *(long*)(newLowerId.Content.Ptr + length + sizeof(long)) = Bits.SwapBytes(GenerateNextEtag()); // now the id will be unique
+            
+            return newLowerId;
+        }
+
+        private const long ConflictedTombstoneIdMarkerLong = 20L;
+        private static readonly string ConflictedTombstoneIdMarkerString = Encoding.UTF8.GetString(BitConverter.GetBytes(ConflictedTombstoneIdMarkerLong));
+
+        private static LazyStringValue UnwrapLowerIdIfNeeded(JsonOperationContext context, string lowerId)
+        {
+            var index = lowerId.IndexOf(ConflictedTombstoneIdMarkerString, StringComparison.OrdinalIgnoreCase);
```

Comment:
We `lowerId` be `LSV` instead of string? 

Then instead of calling `IndexOf` we could check if at position `lowerId.Size - 2*sizeof(long)` we have `ConflictedTombstoneIdMarkerString`?